source/framework/test/src/FrameworkCore.cxx

```diff
@@ -37,3 +39,26 @@ TEST(FrameworkCore, TRestRun) {
     EXPECT_TRUE(run.GetRunDescription() == "This is a test for TRestRun");
     EXPECT_TRUE(run.GetVerboseLevelString() == "debug");
 }
+
+TEST(FrameworkCore, TRestMetadata) {
+    // Check required files exist
+    EXPECT_TRUE(fs::exists(BASIC_TRESTMETADATA_RML));
+    
+    // Create new TRestMetadata class
+    class metadataTestClass: public TRestMetadata{
```

Comment:
Any REST class should be starting by `TRest`

Reply:
> Any REST class should be starting by `TRest`

but this is not a class, its just a cpp file

Reply:
well, it is a class, as it is stated in front, I agree we could make an exception inside tests, but I though it was not possible that the XML header and the class name would be different, it is this possible? @nkx?

Reply:
what are you saying is a class? I'm not following you @jgalan

Reply:
I don't mind the naming convention, we can change it if you prefer. Just I thought it was better not to use REST naming convenctions to avoid confussion with any real REST classes. 
This is a fake one created to be able to test `GetParameter` function from TRestMetadata. It is a pure abstract class so we can't call it, we can only inherit from it.

Related to the name of the metadata section, I can modify it to coincide with name of the class, maybe is easier to read. But `meta.LoadConfigFromFile(BASIC_TRESTMETADATA_RML, "metadata");` is telling where has to look for the parameters to initialize `metadataTestClass` (the XML section called "metadata") so doesn't seem to have any problem.

Reply:
Another thing is if the test is useful or not. This was my fist try with tests and it worked, so far so good. Of couse more useful tests could be done.

Reply:
Right, if it is a fake, I agree it might be better to do not follow the naming conventions.

On the other hand, the natural behaviour should be that xml entry and c++ metadata class got the same name.

I thought `TRestMyMetadata::LoadFromConfigFile( filename, "myName")` would retrieve a RML definition as follows:

`<TRestMyMetadata name="myName" />` from `filename`, and not `<myName ... />` thats why I am worried. By the fact that this is not imposed.

Reply:
> This was introduced by @DavidDiezIb and I think its a great idea to test a class such as `TRestMetadata`.

I am not debating that

Reply:
Yes, that's right, maybe we should remove the option to select the section by hand in `LoadFromConfigFile`. I don't know if this is useful somewhere.

Reply:
Yes, I didn't know that was possible, we never use it that way, isn't @nkx?

```
Int_t TRestMetadata::LoadConfigFromFile(string cfgFileName, string sectionName) {
    fConfigFileName = cfgFileName;
    if (TRestTools::fileExists(fConfigFileName)) {
        if (sectionName == "") {
            sectionName = this->ClassName();
        }
```

So probably we should remove the second argument of that method, and just define a local variable `sectionName=this->ClassName()`?

Reply:
Exactly, I agree.

Reply:
However I am a bit surprised, I thought the second argument was the given name, so that if I got two `TRestMyMetadata` definitions I am able to select the one I am interested in:

```
<TRestMyMetadata name="md1" />

<TRestMyMetadata name="md2" />
```

So that the constructor would pick up the right definition using:
```
TRestMyMetadata *md = new TRestMyMetadata("file.rml", "md1" );
md->LoadConfigFromFile( "file.rml", "md1");
```

At least thats how I am using it inside for example `TRestAxionMagneticField` and it works.

I guess it is because either option is posible as from the definition `NameOrDeclare` at:

```
TiXmlElement* TRestMetadata::GetElementFromFile(std::string cfgFileName, std::string NameOrDecalre) 
```

But why do we need `Declare` @nkx?